bigquery/src/test/java/org/apache/iceberg/gcp/bigquery/BigQueryTableOperationsTest.java

```diff
@@ -0,0 +1,274 @@
+                    .setParameters(ImmutableMap.of(METADATA_LOCATION_PROP, "a/new/location")));
+
+    reset(bigQueryMetaStoreClient);
+    // Two invocations, for loadTable and commit.
+    when(bigQueryMetaStoreClient.load(TABLE_REFERENCE))
+        .thenReturn(tableWithEtag, tableWithNewMetadata);
+
+    org.apache.iceberg.Table loadedTable = bigQueryMetastoreCatalog.loadTable(SPARK_TABLE_ID);
+
+    when(bigQueryMetaStoreClient.update(any(), any())).thenReturn(tableWithEtag);
+    assertThatThrownBy(
+            () -> loadedTable.updateSchema().addColumn("n", Types.IntegerType.get()).commit())
+        .isInstanceOf(CommitFailedException.class)
+        .hasMessageContaining("is not same as the current table metadata location");
+  }
+
+  @Test
+  public void testCreateTableCommitSucceeds() throws Exception {
+    var testTable = createTestTable();
+    when(bigQueryMetaStoreClient.create(any(Table.class))).thenReturn(testTable);
+    when(bigQueryMetaStoreClient.load(
+            new DatasetReference().setProjectId(GCP_PROJECT).setDatasetId(DATASET_ID)))
+        .thenReturn(
+            new Dataset()
+                .setExternalCatalogDatasetOptions(
+                    new ExternalCatalogDatasetOptions()
+                        .setDefaultStorageLocationUri("build/db_folder")));
+
+    Schema schema = testSchema();
+    bigQueryMetastoreCatalog.createTable(SPARK_TABLE_ID, schema, PartitionSpec.unpartitioned());
+  }
+
+  /** Creates a test table to have Iceberg metadata files in place. */
+  private Table createTestTable() throws Exception {
+    when(bigQueryMetaStoreClient.load(TABLE_REFERENCE))
+        .thenThrow(new NoSuchTableException("error message getTable"));
+    return createTestTable(tempFolder, bigQueryMetastoreCatalog, TABLE_REFERENCE);
+  }
+
+  public static Table createTestTable(
+      File tempFolder,
+      BigQueryMetastoreCatalog bigQueryMetastoreCatalog,
+      TableReference tableReference)
+      throws IOException {
+    Schema schema = testSchema();
+    TableIdentifier tableIdentifier =
+        TableIdentifier.of(tableReference.getDatasetId(), tableReference.getTableId());
+    String tableDir = tempFolder.toPath().resolve(tableReference.getTableId()).toString();
+
+    bigQueryMetastoreCatalog
+        .buildTable(tableIdentifier, schema)
+        .withLocation(tableDir)
+        .createTransaction()
+        .commitTransaction();
+
+    Optional<String> metadataLocation = metadataFilePath(tableDir);
+    assertThat(metadataLocation).isPresent();
+    return new Table()
+        .setTableReference(tableReference)
+        .setExternalCatalogTableOptions(
+            new ExternalCatalogTableOptions()
+                .setStorageDescriptor(new StorageDescriptor().setLocationUri(tableDir))
+                .setParameters(
+                    Collections.singletonMap(METADATA_LOCATION_PROP, metadataLocation.get())));
+  }
+
+  private static Schema testSchema() {
```

Comment:
can be defined as `private static final` instead of using a method

Reply:
Done